client/client.go

```diff
@@ -413,7 +417,57 @@ func (client *Client) doSync(method, path string, query url.Values, headers map[
 	return client.doSyncWithOpts(method, path, query, headers, body, v, nil)
 }
 
+// checkMaintenanceJSON checks if there is a maintenance.json file written by
+// snapd the daemon that positively identifies snapd as being unavailable due to
+// maintenance, either for snapd restarting itself to update, or rebooting the
+// system to update the kernel or base snap, etc. If there is ongoing
+// maintenance, then the maintenance object on the client is set appropriately.
+// note that currently checkMaintenanceJSON does not return non-nil errors, so
+// if the file is missing or corrupt or empty, nothing will happen
+func (client *Client) checkMaintenanceJSON() error {
+	f, err := os.Open(dirs.SnapdMaintenanceFile)
+	// just continue if we can't read the maintenance file
+	if err != nil {
+		return nil
+	}
+
```

Comment:
Is there a "defer f.Close()" missing here?

Reply:
yes there is, :facepalm: